### PR TITLE
Add Examples API

### PR DESCRIPTION
**Pull request by msweeneydev**
This PR is based heavily on #350 from @amio, however with updates for zero configuration, it also removes the large diff due to the overhaul of the repository since PR was originally opened.

With this PR, every time the repository is updated, a static build will be created and made available at <now-examples-api.zeit.sh>

As per #350 there are numerous benefits to doing this, the main one however is a single endpoint, no longer making a request to GitHub each time. This will be updated shortly for `now init` and also `/new`.

### CODE REVIEW

File: build-api.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *     /_api/manifest.json
+ *
+ *     Download example files by name (e.g. "apollo")
+ *     /_api/download/apollo.tar.gz
+ */
+
+const debuglog = require('util').debuglog('API_BUILDER');
+const { lstatSync, readdirSync, createWriteStream } = require('fs');
+const { join } = require('path');
+const fse = require('fs-extra');
```
Comment (AndyBitz) [inline build-api.js]: I think you can just `const fs = reqiure('fs-extra')` and then call `fs.lstatSync`, `fs. readdirSync` and `fs.createWriteStream`

File: build-api.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+const tar = require('tar-fs');
+
+main();
```
Comment (AndyBitz) [inline build-api.js]:
```suggestion
main()
  .then(() => process.exit(0))
  .catch((err) => {
    console.error(err);
    process.exit(1);
  });
```

Just to make sure it actually exits, sometimes a different process will keep running after `main` exits, though, it is probably unlikely in this case.

**Comment by skllcrn** [comment]
This will require changes to both `front` and `now-cli` in order to use the new and faster API. Holding off until `init.ts` in `now-cli` has been updated, same goes for `use-example-templates.js` and `project.js` in `front`.

**Comment by leo** [comment]
We should retain the API's signature.

**Pull request closed without merging**